source/fe/fe_system.cc

```diff
@@ -147,31 +147,39 @@ namespace internal
     const unsigned int n_components = fe.element_multiplicity(base_no);
     const unsigned int n_dofs_per_cell =
       fe.base_element(base_no).n_dofs_per_cell();
-    for (unsigned int component = 0; component < n_components; ++component)
-      for (unsigned int b = 0; b < n_dofs_per_cell; ++b)
-        {
-          const unsigned int out_index = base_to_system_table[component][b];
-
-          if (base_flags & update_values)
-            for (unsigned int q = 0; q < n_q_points; ++q)
-              output_data.shape_values[out_index][q] =
-                base_data.shape_values[b][q];
-
-          if (base_flags & update_gradients)
-            for (unsigned int q = 0; q < n_q_points; ++q)
-              output_data.shape_gradients[out_index][q] =
-                base_data.shape_gradients[b][q];
 
-          if (base_flags & update_hessians)
-            for (unsigned int q = 0; q < n_q_points; ++q)
-              output_data.shape_hessians[out_index][q] =
-                base_data.shape_hessians[b][q];
-
-          if (base_flags & update_3rd_derivatives)
-            for (unsigned int q = 0; q < n_q_points; ++q)
-              output_data.shape_3rd_derivatives[out_index][q] =
-                base_data.shape_3rd_derivatives[b][q];
-        }
+    auto copy_row = [&](const auto row_in, auto row_out) {
```

Comment:
The only thing you're capturing here is `n_q_point`. Would you be willing to make that explicit?
```suggestion
    auto copy_row = [n_q_point](const auto row_in, auto row_out) {
```

Reply:
We don't need `n_q_points` at all since we can just use the end iterator (which turns out to be slightly faster anyway).